### PR TITLE
Document npm inter-repo dependencies instructions

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Document npm inter-repo dependencies instructions

### CODE REVIEW

File: README-dev.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
 # README GUIDE FOR CELO DEVELOPERS
 
+## Monorepo inter-package dependencies
+
+Many packages depend on other packages within the monorepo. When this happens, follow these rules:
+
+  1. All packages must use **master version** of sibyl packages.
```
Comment (nambrot) [inline README-dev.md]: sibyl? Do you mean siblings?

File: README-dev.md [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+  2. Exception to (1) are packages that represent a GAE/firebase app which must use the last published version.
+  3. To differentiate published vs unpublished version. Master version (in package.json) must end with suffix `-dev` and should not be published.
+  4. If a developer want to publish a version; then after publishing it needs to set master version to next `-dev` version and change all package.json that require on it.
+
+A practical example:
+
+ * In any given moment, `contractkit/package.json#version` field **must** of the form `x.y.z-dev`
```
Comment (nambrot) [inline README-dev.md:18]: What about utils?

**Review by jmrossy — COMMENTED** [review]
Looks good! A celotool script to update the package versions automatically would be great but not a must have for now. Just waiting for fix to the typo Nam pointed out

File: README-dev.md (hunk 1 above)
Comment (mcortesi) [reply README-dev.md]: heheh... sorry

File: README-dev.md (hunk 2 above)
Comment (mcortesi) [reply README-dev.md:18]: same, this is just an example...
or what do you mean?

**Comment by mcortesi** [comment]
> Looks good! A celotool script to update the package versions automatically would be great but not a must have for now. Just waiting for fix to the typo Nam pointed out

Now you have it...

```
mcortesi celo/celo-monorepo% yarn check:packages       
yarn run v1.19.0
$ node ./scripts/check-packages.js
@celo/attestation-service => @celo/contractkit. expected: 0.1.6 got: 0.1.1
@celo/attestation-service => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/celotool => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/celocli => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/celocli => @celo/contractkit. expected: 0.1.6 got: ^0.1.1
@celo/contractkit => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/dappkit => @celo/contractkit. expected: 0.1.6 got: 0.1.5
@celo/dappkit => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/protocol => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/react-components => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/transaction-metrics-exporter => @celo/contractkit. expected: 0.1.6 got: ^0.1.1
@celo/transaction-metrics-exporter => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/verifier => @celo/utils. expected: 0.1.1 got: ^0.1.0
@celo/walletkit => @celo/utils. expected: 0.1.1 got: ^0.1.0
error Command failed with exit code 1.
```

**Review by jmrossy — APPROVED** [review]
Nice!!

**Pull request merged**